lib/eigensolve_quda.cpp

```diff
@@ -61,27 +66,32 @@ namespace quda
     save_prec = eig_param->save_prec;
 
     // Sanity checks
-    if (n_kr <= n_ev) errorQuda("n_kr = %d is less than or equal to n_ev = %d", n_kr, n_ev);
+    if(eig_param->eig_type == QUDA_EIG_TR_LANCZOS || eig_param->eig_type == QUDA_EIG_BLK_TR_LANCZOS) {
+      if (n_kr <= n_ev) errorQuda("n_kr = %d is less than or equal to n_ev = %d", n_kr, n_ev);
+    } else {
+      if (n_kr < n_ev) errorQuda("n_kr = %d is less than n_ev = %d", n_kr, n_ev);
```

Comment:
Is there a reason we can't use one of these two in practice for all eigensolves?

Reply:
No, I will generalise this.

Reply:
I had this here because I had a normal Arnoldi and I wanted to test the case `n_ev = n_kr`.